toolkit/tools/imagegen/installutils/installutils.go

```diff
@@ -1325,9 +1325,37 @@ func configureUserStartupCommand(installChroot *safechroot.Chroot, user configur
 }
 
 func provisionUserSSHCerts(installChroot *safechroot.Chroot, user configuration.User, homeDir string) (err error) {
+	var (
+		pubKeyData []string
+		exists     bool
+	)
 	const squashErrors = false
+	const authorizedKeysTempFilePerms = 0644
+	const authorizedKeysTempFile = "/tmp/authorized_keys"
 
 	userSSHKeyDir := filepath.Join(homeDir, ".ssh")
+	authorizedKeysFile := filepath.Join(homeDir, ".ssh/authorized_keys")
+
+	exists, err = file.PathExists(authorizedKeysTempFile)
+	if err != nil {
+		logger.Log.Warnf("Error accessing %s file : %v", authorizedKeysTempFile, err)
+		return
+	}
+	if !exists {
+		logger.Log.Debugf("File %s does not exist. Creating file...", authorizedKeysTempFile)
+		err = file.Create(authorizedKeysTempFile, authorizedKeysTempFilePerms)
+		if err != nil {
+			logger.Log.Warnf("Failed to create %s file : %v", authorizedKeysTempFile, err)
+			return
+		}
+		defer os.Remove(authorizedKeysTempFile)
```

Comment:
You should move this defer to right after this if/else so that we try deferring a removal if we were forced to truncate an existing one this call.

Reply:
Done, thanks